### ISSUE TITLE
Proposal: Move sharing settings into settings

### ISSUE THREAD

**Issue by MichaelArestad** [Automattic/wp-calypso#11023]
@beaulebens, @rickybanister, and I are proposing to move the sharing settings into the main settings view. This is going to require a few steps to accomplish.

### 1. Combine Publicize and Sharing Buttons into one view

It could look something like this super quick mockup:
![image](https://cloud.githubusercontent.com/assets/1123119/22392927/fb9c6880-e4bb-11e6-8d96-350166e285f6.png)

Now they would only take up a single tab.

### 2. Make space in the Settings tabs

This would be done by moving Import and Export out of the tabs and into the General section. Their current design almost fits in as is:

![image](https://cloud.githubusercontent.com/assets/1123119/22392983/a166c440-e4bc-11e6-84ec-8db81ebd3533.png)

![image](https://cloud.githubusercontent.com/assets/1123119/22392987/aade2fea-e4bc-11e6-8798-d6ab2fac21d5.png)

For Jetpack sites, we would modify them simply to link back to wp-admin's import/export.

### 3. Add Sharing as a tab in Settings

It would look something like this:
![image](https://cloud.githubusercontent.com/assets/1123119/22392974/6c9e4936-e4bc-11e6-9407-d6f86b4ab7a9.png)

and would contain the now combined sharing section.

### Benefits
* would free up precious sidebar space
* would allow JP and Calypso's UI to align even more
* would keep all the settings together
* would prevent the need for JP to have to find a place for sharing settings

**Comment by rickybanister** [comment]
I think this proposal makes sense, but it might be easier to visualize if we do a slightly more cohesive mockup. Seeing import/export clickable cards within the general settings tab, and perhaps expanded in their own sub pages would be nice to see. And also a more polished view of sharing/publicize in the Traffic tab would be nice to see.

In addition to that, we may want to pull in some other folks to see how we might advertise the existence of sharing/publicize outside of settings. Can it be part of the onboarding flow? Can we resurrect the 'post-post' experience a bit so a user that publishes a post without any connected services is invited to connect then and there to get traffic to their post? 

cc @shaunandrews @folletto

**Comment by folletto** [comment]
Heya! This is good! I agree that since we agree on this to move to a more detailed mock. :)

> we may want to pull in some other folks to see how we might advertise the existence of sharing/publicize outside of settings

It's directly linked in the Editor sidebar, so maybe that's enough?
We probably need to check how many connections are made from the sidebar menu.

> an we resurrect the 'post-post' experience a bit so a user that publishes a post without any connected services is invited to connect then and there to get traffic to their post?

YES PLEASE! Thanks for bringing this up :) 
Post post is imho the absolute right place for sharing! :)

**Comment by MichaelArestad** [comment]
@folletto I think it's good to have them in settings as well so a user doesn't have to go to a post/page to configure their sharing if something busts or changes. Def not needed at a sidebar level.

**Comment by MichaelArestad** [comment]
Ask for high fidelity mockups and you shall receive:

#### Sharing section
![image](https://cloud.githubusercontent.com/assets/1123119/22516950/be9d46f0-e864-11e6-9f60-85d6044d6245.png)

#### In WordPress.com site General Settings in Calypso:
![image](https://cloud.githubusercontent.com/assets/1123119/22516916/a42d9ec8-e864-11e6-8375-a8a82873abb7.png)

#### In Jetpack site General Settings in Calypso:
![image](https://cloud.githubusercontent.com/assets/1123119/22517252/cae70b02-e865-11e6-9ed3-33c0a9c5a104.png)


I just copied the copy already in use for the Import/Export links. I think they could perhaps be improved.

@rickybanister @folletto

**Comment by folletto** [comment]
Solid 💯 

> I think it's good to have them in settings as well so a user doesn't have to go to a post/page to configure their sharing if something busts or changes. 

Ah sorry! I didn't meant to suggest to just have it alone, I meant having in settings AND in the Editor should be enough in terms of showing the feature and activating it. :)

**Comment by rickybanister** [comment]
This is so great. Thanks for shedding lots of light here @MichaelArestad 

I think it's time we improve the Jetpack-specific stuff there on the General tab. I know some of that will be going away as a matter of course, but I see that bottom section with Import/Export as the perfect place for Connection Cycling and Disconnection stuff. Sync would likely go elsewhere—perhaps in the Activity page when it's ready. 

But that's all for a different issue.

**Comment by MichaelArestad** [comment]
@rickybanister Yeah. I was half tempted to go down that rabbit hole, but we'll do that after all this stuff.

**Comment by beaulebens** [comment]
I just realized a potential problem with this proposal.

Users with the Role of Author or Editor have (or should have) access to Publicize settings. They do _not_ have access to (I think) anything else under the current Settings menu.

So in the old approach, they would see the Sharing sidebar item, and when they visited it, they'd see the Publicize/connections details, but the Sharing buttons would be hidden.

With this new proposal, I think they would see the Settings sidebar item, and then the only route available to them would be the new Sharing one, and that would only show Publicize settings (not buttons).

So, not a _problem_ per se, just a consideration/point to be made.

**Comment by MichaelArestad** [comment]
@beaulebens They'll still have access to connect/get there from posts also. I think we can get around this problem without too much trouble.

**Comment by shaunandrews** [comment]
Outside of any permission-related issues that need to be solved, this seems "OK." Its a giant scary page, but I'm not sure its more giant-er, or scary-er than before.

That being said, I think the Sharing Buttons settings should live in the customizer.

**Comment by rickybanister** [comment]
I agree with that—we've also discussed doing the same for Related Posts, possibly the Comments label/theme, and maybe there was another thing.

**Comment by MichaelArestad** [comment]
@shaunandrews I agree. I just want a good intermediary step that is hopefully quick to implement.

**Comment by rickybanister** [comment]
Perhaps with @shaunandrews recent work on the settings navigation we could re-explore this?

With @fditrapani's checklist we can introduce people to sharing settings without needing the top-level sidebar visibility of it.

**Comment by shaunandrews** [comment]
Yea, this will likely be tackled with the sidebar work thats happening now.

**Comment by fditrapani** [comment]
I like the idea of putting this in settings. I rarely go to this screen because I set it up when I created my blog and then never had to go back to it. I also really like Shawn's comments about putting the sharing buttons in the customizer. 

From a discoverability perspective, I can see a number of ways to tackle this: As @rickybanister said we could introduce people to it through onboarding, we can send them emails, our default view of the sharing section in post settings allows people to set them up from there. 

![image](https://user-images.githubusercontent.com/6981253/34856228-f995d4fe-f711-11e7-9a18-2e21e1c8eaaf.png)
